src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Tree/Dependencies/Subscriptions/MSBuildDependencies/MSBuildDependencyFactoryBase.cs

```diff
@@ -423,7 +423,8 @@ internal bool TryMakeUnresolved(MSBuildDependency dependency, out MSBuildDepende
         DiagnosticLevel diagnosticLevel = dependency.DiagnosticLevel <= DiagnosticLevel.Warning ? DiagnosticLevel.Warning : dependency.DiagnosticLevel;
         ProjectTreeFlags flags = UpdateTreeFlags(dependency.Id, FlagCache.Get(isResolved: false, dependency.IsImplicit));
         ProjectImageMoniker icon = GetIcon(dependency.IsImplicit, diagnosticLevel);
-        updated = dependency.With(isResolved: false, icon: icon, flags: flags, diagnosticLevel: diagnosticLevel);
+        string caption = GetUnresolvedCaption(dependency.Id, dependency.BrowseObjectProperties);
+        updated = dependency.With(isResolved: false, icon: icon, flags: flags, diagnosticLevel: diagnosticLevel, caption: caption, filePath: dependency.Id);
```

Comment:
This ensures we reset the caption and file path when converting a dependency from resolves to unresolved.